src/components/menus/SnippetMenu.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { Component, PropTypes } from 'react';
+import { IconMenu } from 'material-ui';
+import SnippetList from './SnippetList';
+import OrgSnippetsMenu from './OrgSnippetsMenu';
+import CustomPropTypes from '../../util/custom-prop-types';
+
+const styles = {
+  iconMenu: {
+    height: '100%',
+    alignItems: 'center',
+    display: 'inline-flex',
+    paddingLeft: '10px',
+    paddingRight: '10px',
+  },
+  menuText: {
+    color: 'white',
+  },
+};
+
+class SnippetMenu extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      menuOpen: false,
+      cursorOnMenuText: false,
+    };
+    this.handleSnippetSelected = this.handleSnippetSelected.bind(this);
+    this.handleMenuHovered = this.handleMenuHovered.bind(this);
+    this.handleMenuNotHovered = this.handleMenuNotHovered.bind(this);
+  }
+
+  handleMenuHovered(ev) {
+    ev.stopPropagation();
+    ev.preventDefault();
+    this.setState({ menuOpen: true });
+  }
+
+  handleMenuNotHovered() {
+    this.setState({ menuOpen: false });
+  }
+
+  handleSnippetSelected(snippetOwner, snippetKey) {
+    this.setState({ menuOpen: false });
+    this.props.onSnippetSelected(snippetOwner, snippetKey);
+  }
+
+  render() {
+    const {
+      style,
+      username,
+      userSnippets,
+      orgSnippets,
+      onHoverBackground,
+      borderBottomColor,
+    } = this.props;
+    const {
+      menuOpen,
+    } = this.state;
+
+    const menuOpenStyle = menuOpen ? {
+      background: onHoverBackground,
+      borderBottomWidth: '3px',
+      borderBottomStyle: 'solid',
+      borderBottomColor,
+    } : {
+      background: 'transparent',
+      borderBottomWidth: '3px',
+      borderBottomColor: 'transparent',
+    };
```

Comment:
These should be refactored to be constants, because every time `render()` is called, both objects will be instantiated (even though one of them won't be used). The constants should be outside of the class definition.